### PR TITLE
Use scoped mocha opts file

### PR DESCRIPTION
**Pull request by rejeep**
(Note that this branch builds on #27, but is targeted against master, so before looking at this, merge #27)

This change will use any locally scoped `mocha.opts` file. This happens when there are different types of tests (unit, acceptance, integration) with their own `mocha.opts` file.

### CODE REVIEW

File: mocha.el [hunk 1]
```diff
@@ -76,6 +77,16 @@ From http://benhollis.net/blog/2015/12/20/nodejs-stack-traces-in-emacs-compilati
       (setq i (+ i 1)))
     dir))
 
+(defun mocha-opts-file (mocha-file)
+  "Return path to mocha.opts file for MOCHA-FILE."
```
Comment (scottaj) [inline mocha.el]: This is cool. I actually didn't know mocha supported opts files. I've just been using different .dir-locals files when I need different test configs in the same project.

Comment (rejeep) [inline mocha.el]: Yes and it will always load `test/mocha.opts` even if not specified.

File: mocha.el [hunk 2]
```diff
@@ -85,10 +96,12 @@ If MOCHA-FILE is specified run just that file otherwise run
 MOCHA-PROJECT-TEST-DIRECTORY.
 
 IF TEST is specified run mocha with a grep for just that test."
-  (let ((path (or mocha-file mocha-project-test-directory))
+  (let* ((path (or mocha-file mocha-project-test-directory))
```
Comment (scottaj) [inline mocha.el]: I don't think the `let*` is necessary?

File: mocha.el [hunk 3]
```diff
@@ -85,10 +96,12 @@ If MOCHA-FILE is specified run just that file otherwise run
...
         (target (if test (concat "--grep \"" test "\" ") ""))
         (node-command (concat mocha-which-node (if debug (concat " --debug=" mocha-debug-port) "")))
         (options (concat mocha-options (if debug " -t 21600000"))))
```
Comment (scottaj) [inline mocha.el]: If you're setting the other special option below, maybe this should be pulled out two and both of those should go into another function?

File: mocha.el [hunk 4]
```diff
@@ -85,10 +96,12 @@ If MOCHA-FILE is specified run just that file otherwise run
...
+    (let ((opts-file (mocha-opts-file mocha-file)))
```
Comment (scottaj) [inline mocha.el]: I think this can be collapsed into the other let?

File: test/mocha-test.el [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+     (f-touch integration-test-file)
+     (should (equal (mocha-opts-file unit-test-file) unit-opts-file))
+     (should (equal (mocha-opts-file acceptance-test-file) acceptance-opts-file))
+     (should-not (mocha-opts-file integration-test-file)))))
+
+

+;;;; mocha-generate-command
+
+(ert-deftest mocha-test/mocha-generate-command/return-command-including-mocha-opts-option ()
```
Comment (scottaj) [inline test/mocha-test.el]: Could you add the negative case to this one for the integration test file? that is has no --opts option?

File: mocha.el (hunk 2 above)
Comment (rejeep) [inline mocha.el]: No, was before but not now.

File: mocha.el (hunk 4 above)
Comment (rejeep) [inline mocha.el]: 👍

File: test/mocha-test.el (hunk 5 above)
Comment (rejeep) [inline test/mocha-test.el]: 👍

File: mocha.el (hunk 3 above)
Comment (rejeep) [inline mocha.el]: Perhaps we should refactor this whole function and at the same time write some tests for it as it is such a crucial part? But that would be for another PR.

**Comment by rejeep** [comment]
The tests fail for me locally with _"SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode"_. Do you know if I have to run it with a specific version of Node?

**Comment by rejeep** [comment]
Node 6 solved it.

**Comment by scottaj** [comment]
Yeah, I just put an .nvmrc with node 6 in the sample project directory.

File: mocha.el (hunk 3 above)
Comment (scottaj) [inline mocha.el]: That sounds like a good plan 👍

**Comment by scottaj** [comment]
How do you feel about adding a mocha.opts to the sample project and adding a cuke for this?

**Comment by rejeep** [comment]
> How do you feel about adding a mocha.opts to the sample project and adding a cuke for this?

Jupp, will do that!

I also found a big with help of your Ecukes-tests that I will fix.

**Comment by rejeep** [comment]
I started writing Ecukes tests but soon realised that I don't think we should write many of those. Adding more Node-tests changes the outcome of other features because they run tests for the complete project. Instead I think we should focus on unit tests and only have a few Ecukes tests that make sure all parts run well together.

I did some updates to this branch from your feedback, mind taking another look? Btw, Travis says this PR fails, but it's fine as you can see here https://travis-ci.org/scottaj/mocha.el/builds/132311718, no idea why it doesn't update to the latest run.

**Comment by scottaj** [comment]
This LGTM.

As fare as cukes go, I think we'll have to make the sample project a little more complex, and maybe have multiple test directories to test different cases like CoffeeScript support. But I'm fine with deferring that till we have more major features to integrate.

**Comment by rejeep** [comment]
Merging this then! 👍

**Comment by rejeep** [comment]
And I saw you just did! 😄
